src/solvers/mod.rs

```diff
@@ -230,8 +230,31 @@ pub trait WithInitialSolution {
     fn with_initial_solution(self, solution: impl IntoIterator<Item = (Variable, f64)>) -> Self;
 }
 
+/// A solver than can stop the solving process after some time
+pub trait WithTimeLimit {
+    /// Sets the time limit for the solver
+    fn with_time_limit<T: Into<f64>>(self, seconds: T) -> Self;
+}
+
+/// Information about the status of a solution, such as whether the solution is
+/// optimal
+#[derive(Clone, Copy, Debug)]
+pub enum SolutionStatus {
+    /// The solution is optimal
+    Optimal,
+    /// The solution is not optimal and it was obtained because the time limit
+    /// was reached
+    TimeLimit,
+    /// The solution is not optimal and it was obtained because the gap limit
+    /// was reached
+    GapLimit,
```

Comment:
I would ideally like to do
```suggestion
    GapLimit(f64),
```
but I think the underlying bindings for CBC and SCIP don't expose this data yet.

I mostly think this would be better library design but I honestly don't ever need this myself … so I might just not do it?